src/os/style/circlereader.js

```diff
@@ -42,14 +45,32 @@ os.style.CircleReader.prototype.getOrCreateStyle = function(config) {
     hash = 31 * hash + stroke['id'] >>> 0;
   }
 
+  var color = /** @type {string} */ (config['color']);
+  if (color) {
+    hash = 31 + hash + goog.string.hashCode(color) >>> 0;
+
+    var rgbArray = os.color.toRgbArray(color);
+    if (rgbArray) {
+      opacity = rgbArray[3];
+    }
+  }
+
   if (!this.cache[hash]) {
-    this.cache[hash] = new ol.style.Circle({
+    var style = new ol.style.Circle({
       radius: radius,
       fill: fill,
       stroke: stroke
     });
 
-    this.cache[hash]['id'] = hash;
+    if (opacity != null) {
+      // set the opacity if it's defined, this is used by the WebGL synchronizers and is not passed by the options
+      // up to the parent style class for some reason (???)
```

Comment:
From OL's point of view, the Image style's opacity value isn't needed for the RegularShape style because it has separate fill/stroke colors each with their own opacity. The Icon style does use the value despite having a color property, but icon color is optional to render something where a shape has to have a fill or stroke to be visible.